.github/workflows/ci.yaml

```diff
@@ -1,87 +1,148 @@
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
+      - name: Check that the CI pipeline definition is up-to-date with the dhall configuration
+        run: ci/check-rendered-pipeline-up-to-date.sh
+  dhallFormat:
+    name: dhall-format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v2'
+      - name: '[setup] Install asdf'
+        uses: 'asdf-vm/actions/setup@v1.0.1'
+      - name: '[setup] Install asdf plugins'
+        run: './ci/asdf-add-plugins.sh'
+      - name: '[setup] Install asdf .tool-versions'
+        run: './ci/asdf-install.sh'
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
+      - env:
+          CHECK: 'true'
+        name: Check that dhall files are formatted
+        run: just format-dhall
+  dhallFreeze:
+    name: dhall-freeze
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v2'
+      - name: '[setup] Install asdf'
+        uses: 'asdf-vm/actions/setup@v1.0.1'
+      - name: '[setup] Install asdf plugins'
+        run: './ci/asdf-add-plugins.sh'
+      - name: '[setup] Install asdf .tool-versions'
+        run: './ci/asdf-install.sh'
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
+      - env:
+          CHECK: 'true'
+        name: Check that dhall files are linted properly
+        run: just freeze-dhall
+  dhallLint:
+    name: dhall-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v2'
+      - name: '[setup] Install asdf'
+        uses: 'asdf-vm/actions/setup@v1.0.1'
+      - name: '[setup] Install asdf plugins'
+        run: './ci/asdf-add-plugins.sh'
+      - name: '[setup] Install asdf .tool-versions'
+        run: './ci/asdf-install.sh'
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
+      - env:
+          CHECK: 'true'
+        name: Check that dhall files are linted properly
+        run: just lint-dhall
+  goTest:
+    name: go-test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v2'
+      - name: '[setup] Install asdf'
+        uses: 'asdf-vm/actions/setup@v1.0.1'
+      - name: '[setup] Install asdf plugins'
+        run: './ci/asdf-add-plugins.sh'
+      - name: '[setup] Install asdf .tool-versions'
+        run: './ci/asdf-install.sh'
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
+      - name: go-test
+        run: ci/go-test.sh
+  golangci-lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: 'actions/checkout@v2'
+      - name: '[setup] Install asdf'
+        uses: 'asdf-vm/actions/setup@v1.0.1'
+      - name: '[setup] Install asdf plugins'
+        run: './ci/asdf-add-plugins.sh'
+      - name: '[setup] Install asdf .tool-versions'
+        run: './ci/asdf-install.sh'
+      - name: '[setup] prepare go environment'
+        run: './ci/asdf-setup-go.sh'
```

Comment:
Do we need this for golangci-lint? The official action seems to handle all this https://github.com/golangci/golangci-lint-action#performance